### PR TITLE
Changes in Event Tickets to support Event Tickets Plus custom meta

### PR DESCRIPTION
**Pull request by borkweb**
These are the changes required to allow ET+ to hook into the core plugin to provided custom meta data.

Internal ref: 
* https://central.tri.be/issues/34341
* https://central.tri.be/issues/34342
* https://central.tri.be/issues/34343
* https://central.tri.be/issues/34344
* https://central.tri.be/issues/34346
* https://central.tri.be/issues/34348
* https://central.tri.be/issues/42747
* https://central.tri.be/issues/42811

### CODE REVIEW

File: src/Tribe/Tickets.php [hunk 1]
```diff
@@ -380,6 +380,8 @@ public function has_permission( $post, $data, $nonce_action ) {
 
 			$ticket->provider_class = $this->className;
 
+			do_action( 'tribe_tickets_ticket_add', $post_id, $ticket, $data );
```
Comment (bordoni) [inline src/Tribe/Tickets.php]: :memo: Doc Ref

File: src/Tribe/Tickets.php [hunk 2]
```diff
@@ -539,7 +541,7 @@ public function has_permission( $post, $data, $nonce_action ) {
 			$return['description'] = htmlspecialchars_decode( $return['description'] );
 
 			ob_start();
-			$this->do_metabox_advanced_options( $post_id, $ticket_id );
+			do_action( 'tribe_events_tickets_metabox_advanced', $post_id, $ticket_id );
```
Comment (bordoni) [inline src/Tribe/Tickets.php]: :memo: Doc Ref

File: src/admin-views/meta-box.php [hunk 3]
```diff
@@ -168,7 +168,8 @@ class="ticket_field">
 					</td>
 				</tr>
 
-				<?php do_action( 'tribe_events_tickets_metabox_advanced', get_the_ID(), null ); ?>
+				<?php
+				do_action( 'tribe_events_tickets_metabox_advanced', get_the_ID(), null ); ?>
```
Comment (bordoni) [inline src/admin-views/meta-box.php]: :memo: Doc Ref

File: src/Tribe/Attendees_Table.php [hunk 4]
```diff
@@ -193,8 +218,9 @@ public function single_row( $item ) {
 		echo '<tr class="' . sanitize_html_class( $row_class ) . esc_attr( $checked ) . '">';
 		$this->single_row_columns( $item );
 		echo '</tr>';
-	}
 
+		do_action( 'event_tickets_attendees_table_after_row', $item );
```
Comment (bordoni) [inline src/Tribe/Attendees_Table.php]: :memo: Doc Ref

File: src/Tribe/Attendees_Table.php [hunk 5]
```diff
@@ -163,6 +160,34 @@ public function column_order_status( $item ) {
...
+		?>
+		<div class="event-tickets-ticket-name">
+			<?php echo esc_html( $item['ticket'] ); ?>
+		</div>
+		<div class="event-tickets-ticket-purchaser">
+			<?php esc_html_e( 'Purchased by:', 'event-tickets' ); ?> <?php echo esc_html( $item['purchaser_name'] ); ?> (<?php echo esc_html( $item['purchaser_email'] ); ?>)
+		</div>
+		<?php
+
+		do_action( 'event_tickets_attendees_table_ticket_column', $item );
```
Comment (bordoni) [inline src/Tribe/Attendees_Table.php]: :memo: Doc Ref

**Comment by borkweb** [comment]
This one is also ready for another look-see.

**Comment by bordoni** [comment]
:100:

**Pull request merged**